mobile/src/screens/LoadingScreen.tsx

```diff
@@ -0,0 +1,56 @@
+import { useNavigation } from "@react-navigation/native";
+import React, { useContext, useState } from "react";
+import { Text, View, Image } from "react-native";
+
+import stemettImage from "../../assets/Stemett.png";
+import Button from "../components/Button";
+import TwoButtonPopup from "../components/popups/TwoButtonPopup";
+import { SocketContext } from "../contexts/SocketContext";
+import { UserContext } from "../contexts/UserContext";
+import { AppStylesheet } from "../styles/AppStylesheet";
+
+const LoadingScreen = () => {
+  const navigation = useNavigation();
+  const socket = useContext(SocketContext);
+  const { userState } = useContext(UserContext);
+  const [stopPopup, setStopPopup] = useState(false);
+  let playerColor = "";
+
+  const quitSearch = () => {
+    setStopPopup(false);
+    navigation.navigate("HomeScreen");
+    socket.emit("quit searching");
+  };
+
+  socket.once("successful assign", (color: string) => {
```

Comment:
Let's put these into a use effect and just do socket.on to be consistent with the other places we did this.

Reply:
added use effect